packages/backend/src/data/migrations/20210317182303-apidef-endpoint-unique.ts

```diff
@@ -0,0 +1,26 @@
+import { Db } from 'mongodb';
+import { getArrayRepeats } from '../../common/tool';
+
+const index: any = { endpoint: 1 };
+
+export const up = async (db: Db): Promise<void> => {
+  const apiDefsCollection = await db.collection('apidefs');
+  const apiDefs: {
+    endpoint: string;
+  }[] = await apiDefsCollection.find().toArray();
+  const repeats = getArrayRepeats(apiDefs.map((apiDef) => apiDef.endpoint));
+  if (repeats.length) {
+    throw new Error(
+      `Cannot create index as apiDef.endpoint has repeats: ${repeats}`
```

Comment:
We are looking for non-unique apiDef.endpoint values, right? If yes, then we should provide an actionable error message:
`Cannot create an index as apiDef.endpoint "${apiDef.name} -> ${apiDef.endpoint}" is non unique and has duplicate values.`. Ideally, we should provide a list of apiDefs that we have to fix:
`Cannot create an index as apiDef.endpoint "${apiDef.endpoint}" is non-unique and is used in the following apiDefs: ${duplicateApiDefs.names}`.